addons/bus/static/tests/im_status.test.js

```diff
@@ -0,0 +1,123 @@
+import { AWAY_DELAY as ACTUAL_AWAY_DELAY } from "@bus/im_status_service";
+import { test, expect } from "@odoo/hoot";
+import {
+    getService,
+    makeMockEnv,
+    patchWithCleanup,
+    serverState,
+} from "@web/../tests/web_test_helpers";
+import { step, assertSteps, defineBusModels, startServer } from "./bus_test_helpers";
+import { advanceTime, animationFrame } from "@odoo/hoot-mock";
+import { browser } from "@web/core/browser/browser";
+
+defineBusModels();
+
+// Delays can slighly differ since time is not frozen. Let's tolerate 1000ms
+// of difference.
+const TOLERANCE = 1000;
+const AWAY_DELAY = ACTUAL_AWAY_DELAY + TOLERANCE;
+function assertAlmostEqual(a, b) {
+    expect(Math.abs(a - b) < TOLERANCE).toBe(true, {
+        message: `${a} and ${b} should be almost equal`,
+    });
+}
+
+test("update presence if IM status changes to offline while this device is online", async () => {
+    const pyEnv = await startServer();
+    await makeMockEnv();
+    patchWithCleanup(getService("bus_service"), { send: (type) => step(type) });
+    await assertSteps(["update_presence"]);
+    pyEnv["bus.bus"]._sendone(serverState.partnerId, "bus.bus/im_status_updated", {
+        im_status: "offline",
+        partner_id: serverState.partnerId,
+    });
+    await assertSteps(["update_presence"]);
+});
+
+test("update presence if IM status changes to away while this device is online", async () => {
+    const pyEnv = await startServer();
+    await makeMockEnv();
+    patchWithCleanup(getService("bus_service"), { send: (type) => step(type) });
+    patchWithCleanup(getService("presence"), { getLastPresence: () => new Date().getTime() });
+    await assertSteps(["update_presence"]);
+    pyEnv["bus.bus"]._sendone(serverState.partnerId, "bus.bus/im_status_updated", {
+        im_status: "away",
+        partner_id: serverState.partnerId,
+    });
+    await assertSteps(["update_presence"]);
+});
+
+test("do not update presence if IM status changes to away while this device is away", async () => {
+    const pyEnv = await startServer();
+    await makeMockEnv();
+    patchWithCleanup(getService("bus_service"), { send: (type) => step(type) });
+    patchWithCleanup(getService("presence"), {
+        getLastPresence: () => new Date().getTime() - AWAY_DELAY,
+    });
+    await assertSteps(["update_presence"]);
+    pyEnv["bus.bus"]._sendone(serverState.partnerId, "bus.bus/im_status_updated", {
+        im_status: "away",
+        partner_id: serverState.partnerId,
+    });
+    await animationFrame();
```

Comment:
Was it an animation frame initially? I don't think so as it has nothing to do with animation but rather with event dispatching delay. `tick()` looks more appropriate?